Breaking Change: Separate App and Sanity Studio containers + additional studio configuration

FrancescaFr opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)

Please carefully read the code of conduct before creating a pull request
 👉 https://schroedinger-hat.org/page/code-of-conduct
-->

### 🔗 Linked issue

<!-- Please ensure there is an open issue and mention its number as #123 -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [X] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [X] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

1. Separated the Sanity Studio portion of the application from the NextJS frontend. 
    - Separate folders, package.json, .env and dockerfiles
    - Created two separate services in Docker Compose File (with their own localports)
    - remove `/studio` path from frontend (now hosted at its own url)
2. Configured dev environment to reflect test database only
    - new .env variable to point to test database (referenced by local nextjs app)
    - local studio editor likewise only sees development database
    - deployed studio configuration shows production and test database
3. Added mechanism for pre-populating empty test database with a pre-set dataset 
    - currently populates with minimal data upon `docker compose up --build`

Concerns for reviewer:
- too many modifications in the same PR
- incorrect separation of environments (possible breaking changes))
- incorrect port configuration
- breaks in production

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

FrancescaFr commented:
This was my interpretation of "Create a docker-compose image to have a local sanity installation" from trello - to me it made sense to separate the next app from the studio completely (ex. having schema defined in two places seemed bad). I put both services in the original docker-compose for now, but they can easily be separated. I eliminated the '/studio' path, but it could be re-routed to a deployed studio in production. I did not see a clear way to have a fully local sanity instance (ex. local database, etc.) if that was the original intention.

FrancescaFr commented:
Okay - I put the whole sanity studio setup back inside of app directory mostly how it was so its files can be referenced by the `/studio` route (didn't realize it was accessed from front-end during production; thought it was deployed to sanity.io separately... whoops!). 

The studio can now be launched independently (and deployed to studio.io) or alongside the frontend with a seperate docker compose file. `docker compose -f docker-compose.sanity.yml up --build`

There is an additional burden of separate  `.env` files that the user must generate for the studio and separate `sanity.config` files for the stand-alone and nextjs embedded studio views.

Starting from Frontend:
<img width="1164" height="698" alt="Screenshot from 2025-07-12 10-15-03" src="https://github.com/user-attachments/assets/08c04abb-ef55-4555-a0c7-cefd4d34842d" />

Dev Mode: (Shows only dev dataset)
<img width="1164" height="698" alt="Screenshot from 2025-07-12 10-20-56" src="https://github.com/user-attachments/assets/00aaf4fb-d7ff-4a6b-b399-095a3894aa2c" />


Prod Mode: (Shows prod dataset on frontend, prod & dev in studio)
<img width="1048" height="599" alt="image" src="https://github.com/user-attachments/assets/6b51113d-8855-4157-bf94-9372b61fccf6" />

<img width="1164" height="698" alt="Screenshot from 2025-07-12 10-15-24" src="https://github.com/user-attachments/assets/77910ba4-7669-42c8-9502-b876cacd6b36" />

TheJoin95 commented on app/Dockerfile:
```diff
@@ -14,4 +14,4 @@ RUN npm run build
 # Use dev command for development
-CMD ["npm", "run", "dev"] 
\ No newline at end of file
+# CMD ["npm", "run", "dev"] 
```
is this still needed ?

TheJoin95 commented:
Overall it's looking good, can you update the readme about how to setup the dev env ?
And eventually also how to connect the local to the prod one, with a big note that every change will be reflected to the prod dataset

FrancescaFr commented:
Okay - I re-tested and updated the README to reflect the changes.

I did notice that as I tested the production deployment, I was not able to trigger the `production.Dockerfile` with the following command:
`ENVIRONMENT=production docker compose up --build` instead I had to use `NODE_ENV=production docker compose up --build` 

I updated the README to reflect this, but perhaps it is an issue on my end (this command also produced an error during build that I am not able to resolve, if you have some insight).

```bash
CACHED [builder 4/4] RUN npm run build                                                                                                                                      0.0s
 => CANCELED [runner 4/6] COPY --from=builder /app/public ./public                                                                                                              0.0s
 => ERROR [runner 5/6] COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./                                                                                       0.0s
------
 > [runner 5/6] COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./:
------
production.Dockerfile:26

--------------------

  24 |     

  25 |     COPY --from=builder /app/public ./public

  26 | >>> COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./

  27 |     COPY --from=builder --chown=nextjs:nodejs /app/.next/static ./.next/static

  28 |     

--------------------

failed to solve: failed to compute cache key: failed to calculate checksum of ref 40e5de41-ec03-4d3b-ae1b-84cd383ac933::q3br5u6oc3buhv2zooxszmkhm: "/app/.next/standalone": not found
```

TheJoin95 commented:
Weird as it's doing the npm run build right before?
Can you check and try to fix again by checking with some logs that after the npm run build step the files are there?

and eventually update accordengly

TheJoin95 commented:
I see the issue now: https://nextjs.org/docs/pages/api-reference/config/next-config-js/output
we're missing the output prop in the next.config.ts file

can you try to add it following the doc? I believe that prod docker is broken also on the main branch
I'm able to reproduce it locally as well, I guess that we can fix by updating the next configuration

FrancescaFr commented:
@TheJoin95 
okay - updated production dockerfile. seems to be working now.
